core/src/hir/types.rs

```diff
@@ -114,6 +115,7 @@ impl<P: TyPosition> Type<P> {
                     .map(|lt| std::slice::from_ref(lt).iter().copied())
                     .unwrap_or([].iter().copied()),
             ),
+            // TODO the Callback case
```

Comment:
thought: callbacks basically break this form of lifetime computation anyway since they contain contravariant lifetimes. so this is actually probably okay as long as we're banning borrows in callbacks

Reply:
ok fair enough!